### PR TITLE
(Public swagger update) Add TeradataSource,TeradataPartitionSettings,TeradataTableDataset,TeradataTableDatasetTypeProperties

### PR DESCRIPTION
**Pull request by Yiliu-microsoft**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [ ] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [ ] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

File: specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/entityTypes/Pipeline.json
Comment (nschonni) [inline specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/entityTypes/Pipeline.json]:
```suggestion
          "description": "The maximum value of column specified in partitionColumnName that will be used for proceeding range partitioning. Type: string (or Expression with resultType string)."
```

File: specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/entityTypes/Pipeline.json
Comment (nschonni) [inline specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/entityTypes/Pipeline.json]:
```suggestion
          "description": "The minimum value of column specified in partitionColumnName that will be used for proceeding range partitioning. Type: string (or Expression with resultType string)."
```

**Review by nschonni — COMMENTED** [review]
specificed -> specified

**Comment by xiaoyingQQ** [comment]
@nschonni , could you help approve this PR? The comment is resolved.

**Comment by nschonni** [comment]
@xiaoyingLJ sorry, i'm not an MS employee, it'll need to be one of the actual repo admins

**Comment by xiaoyingQQ** [comment]
@jhendrixMSFT ,  could you help merge this PR?

**Comment by jhendrixMSFT** [comment]
@xiaoyingQQ ARM needs to approve the change before I can merge.

**Comment by xiaoyingLJ** [comment]
How long does it need for ARM approve? This PR is waiting for some days. We want this change is merged in this week to catch our ETA of this feature. Could you help me to  expedite this PR?

Great thanks
Xiaoying

**Comment by jhendrixMSFT** [comment]
@xiaoyingLJ please read the initial comment in the PR, it contains instructions for how to expedite the ARM review process.

**Review by jhendrixMSFT — APPROVED** [review]
Waiting for ARM review, will merge once we have that.

File: specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/entityTypes/Dataset.json
Comment (majastrz) [inline specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/entityTypes/Dataset.json]: >Teradata [](start = 5, length = 8)

I am really confused about these changes. The new definitions aren't really referenced in any paths, so this will result in no change to the API contract. Autorest might generate types for all the new definitions, but the clients won't get any new methods to utilize them.

What is the intent here? If you are planning to add paths later, will you be introducing a new API version?

File: specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/entityTypes/Dataset.json
Comment (majastrz) [inline specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/entityTypes/Dataset.json]: >object [](start = 15, length = 6)

I see that you're modeling a lot of properties as object without referencing a definition. In the c# generated SDK, the property type will be `object`. Is that intended? How do you pass an expression via the C# SDK? What kind of expression are these also?

**Review by majastrz — COMMENTED** [review]
I took a look at the changes and added a couple of questions. Can you take a look? We can discuss at the meeting you scheduled for today as well.

File: specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/entityTypes/Dataset.json
Comment (majastrz) [reply specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/entityTypes/Dataset.json]: I missed that we're using discriminator value.

---
In reply to: [295493306](https://github.com/Azure/azure-rest-api-specs/pull/6346#discussion_r295493306) [](ancestors = 295493306)

File: specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/entityTypes/Dataset.json
Comment (majastrz) [reply specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/entityTypes/Dataset.json]: Discussed on a call. C# SDK has a class that users can new up and pass in for the property value. Also backend supports multiple types.

---
In reply to: [295494222](https://github.com/Azure/azure-rest-api-specs/pull/6346#discussion_r295494222) [](ancestors = 295494222)

File: specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/entityTypes/Dataset.json
Comment (majastrz) [reply specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/entityTypes/Dataset.json]: The users know that they're expected to update to latest SDK to use newest types of data sets.

---
In reply to: [295580149](https://github.com/Azure/azure-rest-api-specs/pull/6346#discussion_r295580149) [](ancestors = 295580149,295493306)

**Review by majastrz — APPROVED** [review]
Looks good to me.

**Pull request merged**
